### PR TITLE
feat: fetch & show notes for client and group, fixes #543

### LINKED ISSUE #543: Feature Fetch Notes
**Summary:** 

There is no feature to show Notes of client group savings and loan.
Add A Note Fragment that will be the generic type and will handle all type of note like client group savings and loan.

API: https://demo.openmf.org/api-docs/apiLive.htm#notes

**Steps to reproduce:** 

There is no feature of notes in Client Savings Group and Loan Account menu item.

**Expected behavior:** 

Add Toolbar menu item in ClientListFragment and GroupListFragment and SavingAccountSummaryFragment and LoanAccountSummaryFragment that will make fragment transaction to NoteFragment.

**Observed behavior:** 

There is no feature to show notes.

### PR DESCRIPTION
**Pull request by rahulbabbar1**
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

**Added option to show Notes of the Client and group in the toolbar menu for ClientDetailsFragment and GroupDetailsFragment.**

### CODE REVIEW

File: mifosng-android/src/main/java/com/mifos/api/BaseApiManager.java [hunk 1]
```diff
@@ -54,6 +55,7 @@
     private static StaffService staffApi;
     private static SurveyService surveyApi;
     private static RunReportsService runreportsService;
+    private static NoteItemService noteItemService;
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/api/BaseApiManager.java]: Use Proper Naming convention. This should be like `NoteService`

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/adapters/NoteItemAdapter.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import javax.inject.Inject;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+/**
+ * Created by rahul on 4/3/17.
+ */
+
+public class NoteItemAdapter extends RecyclerView.Adapter<NoteItemAdapter.ViewHolder> {
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/adapters/NoteItemAdapter.java]: Class name should be `NoteAdapter`

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/noteitem/NoteItemFragment.java [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+import javax.inject.Inject;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+/**
+ * Created by rahul on 4/3/17.
+ */
+public class NoteItemFragment extends MifosBaseFragment implements NoteItemMvpView,
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/noteitem/NoteItemFragment.java]: This class name should be `NoteFragment`

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/noteitem/NoteItemFragment.java [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        SwipeRefreshLayout.OnRefreshListener {
+
+    @BindView(R.id.rv_note)
+    RecyclerView rvNote;
+
+    @BindView(R.id.progressbar_note)
+    ProgressBar pbNote;
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/noteitem/NoteItemFragment.java]: No need to add the ProgressBar in UI, Use MifosBaseFragment Dynamic ProgressBar.
You can access using the ` showMifosProgressBar();` and hide by `hideMifosProgressBar();`

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/noteitem/NoteItemFragment.java [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+    SwipeRefreshLayout swipeRefreshLayout;
+
+    @BindView(R.id.tv_error)
+    TextView tvError;
+
+    @BindView(R.id.iv_error)
+    ImageView ivError;
+
+    @BindView(R.id.ll_error)
+    LinearLayout ll_error;
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/noteitem/NoteItemFragment.java]: Failed name should be `llError`

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/noteitem/NoteItemFragment.java [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+        } else {
+            pbNote.setVisibility(View.GONE);
+            swipeRefreshLayout.setRefreshing(show);
+        }
+    }
+
+    @Override
+    public void onDestroyView() {
+        super.onDestroyView();
+        noteItemPresenter.detachView();
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/noteitem/NoteItemFragment.java]: When you will add MifosBaseFragment Progressbar then do not forget to add `hideMifosProgressBar();` here.

File: mifosng-android/src/main/java/com/mifos/objects/noncore/NoteItem.java [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+import android.os.Parcelable;
+
+import com.mifos.api.local.MifosBaseModel;
+
+/**
+ * Created by rahul on 4/3/17.
+ */
+public class NoteItem extends MifosBaseModel implements Parcelable {
+
+    private int id;
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/objects/noncore/NoteItem.java]: Annotate fields with ` @SerializedName("")` and remove private identifier.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/noteitem/NoteItemMvpView.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+import java.util.List;
+
+/**
+ * Created by rahul on 4/3/17.
+ */
+public interface NoteItemMvpView extends MvpView {
+
+    void showUserInterface();
+
+    void showNoteItem(List<NoteItem> noteItem);
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/noteitem/NoteItemMvpView.java]: Method name should be `showNote(List<NoteItem> noteItem)`

**Comment by therajanmaurya** [comment]
@rahulbabbar1 Please make the requested changes.

**Comment by rahulbabbar1** [comment]
@therajanmaurya i will fix them soon.

**Comment by rahulbabbar1** [comment]
@therajanmaurya I have fixed the requested changes. Please check.

**Comment by therajanmaurya** [comment]
Good Work 👍

**Pull request merged**